geomstats/geometry/spd_matrices_space.py

```diff
@@ -453,20 +468,34 @@ def exp(self, tangent_vec, base_point):
         if n_base_points == 1:
             base_point = gs.tile(base_point, (n_tangent_vecs, 1, 1))
 
-        sqrt_base_point = gs.linalg.sqrtm(base_point)
-
-        inv_sqrt_base_point = gs.linalg.inv(sqrt_base_point)
-
-        tangent_vec_at_id = gs.matmul(inv_sqrt_base_point,
-                                      tangent_vec)
-        tangent_vec_at_id = gs.matmul(tangent_vec_at_id,
-                                      inv_sqrt_base_point)
-        exp_from_id = gs.linalg.expm(tangent_vec_at_id)
+        if power_affine == 1:
+            sqrt_base_point = gs.linalg.sqrtm(base_point)
+            inv_sqrt_base_point = gs.linalg.inv(sqrt_base_point)
+            exp = self._aux_exp(tangent_vec, sqrt_base_point,
+                                inv_sqrt_base_point)
+        else:
+            modified_tangent_vec = self.space.differential_power(power_affine,
+                                                                 tangent_vec,
+                                                                 base_point)
+            power_sqrt_base_point = gs.linalg.powerm(base_point,
+                                                     power_affine/2)
+            power_inv_sqrt_base_point = gs.linalg.inv(power_sqrt_base_point)
+            exp = self._aux_exp(modified_tangent_vec, power_sqrt_base_point,
+                                power_inv_sqrt_base_point)
+            exp = gs.linalg.powerm(exp, 1/power_affine)
+
+        if ndim == 2:
```

Comment:
Nice! This is what we will do with decorators for all functions (hopefully, soon)